// @openrouter/sdk/esm/react-query/modelsListEmbeddings.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useQuery, useSuspenseQuery, } from "@tanstack/react-query";
import { modelsListEmbeddings } from "../funcs/modelsListEmbeddings.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useOpenRouterContext } from "./_context.js";
/**
 * List all embeddings models
 *
 * @remarks
 * Returns a list of all available embeddings models and their properties
 */
export function useModelsListEmbeddings(options) {
    const client = useOpenRouterContext();
    return useQuery({
        ...buildModelsListEmbeddingsQuery(client, options),
        ...options,
    });
}
/**
 * List all embeddings models
 *
 * @remarks
 * Returns a list of all available embeddings models and their properties
 */
export function useModelsListEmbeddingsSuspense(options) {
    const client = useOpenRouterContext();
    return useSuspenseQuery({
        ...buildModelsListEmbeddingsQuery(client, options),
        ...options,
    });
}
export function prefetchModelsListEmbeddings(queryClient, client$) {
    return queryClient.prefetchQuery({
        ...buildModelsListEmbeddingsQuery(client$),
    });
}
export function setModelsListEmbeddingsData(client, data) {
    const key = queryKeyModelsListEmbeddings();
    return client.setQueryData(key, data);
}
export function invalidateAllModelsListEmbeddings(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: ["@openrouter/sdk", "Models", "listEmbeddings"],
    });
}
export function buildModelsListEmbeddingsQuery(client$, options) {
    return {
        queryKey: queryKeyModelsListEmbeddings(),
        queryFn: async function modelsListEmbeddingsQueryFn(ctx) {
            const sig = combineSignals(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return unwrapAsync(modelsListEmbeddings(client$, mergedOptions));
        },
    };
}
export function queryKeyModelsListEmbeddings() {
    return ["@openrouter/sdk", "Models", "listEmbeddings"];
}
