// @openrouter/sdk/esm/models/chatmessagecontentitemvideo.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const VideoUrl$inboundSchema = z.object({
    url: z.string(),
});
/** @internal */
export const VideoUrl$outboundSchema = z
    .object({
    url: z.string(),
});
export function videoUrlToJSON(videoUrl) {
    return JSON.stringify(VideoUrl$outboundSchema.parse(videoUrl));
}
export function videoUrlFromJSON(jsonString) {
    return safeParse(jsonString, (x) => VideoUrl$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'VideoUrl' from JSON`);
}
/** @internal */
export const ChatMessageContentItemVideo$inboundSchema = z.object({
    type: z.literal("input_video"),
    video_url: z.lazy(() => VideoUrl$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "video_url": "videoUrl",
    });
});
/** @internal */
export const ChatMessageContentItemVideo$outboundSchema = z.object({
    type: z.literal("input_video"),
    videoUrl: z.lazy(() => VideoUrl$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        videoUrl: "video_url",
    });
});
export function chatMessageContentItemVideoToJSON(chatMessageContentItemVideo) {
    return JSON.stringify(ChatMessageContentItemVideo$outboundSchema.parse(chatMessageContentItemVideo));
}
export function chatMessageContentItemVideoFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatMessageContentItemVideo$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatMessageContentItemVideo' from JSON`);
}
